src/gallia/commands/discover/doip.py

```diff
@@ -545,3 +543,56 @@ async def run_udp_discovery(self) -> list[tuple[str, int]]:
             )
 
         return found
+
+
+class FastDoIPConnection(DoIPConnection):
+    # This code is pretty much copied from the original except for the special treatment of DiagnosticMessages which
```

Comment:
Obviously, introducing a second queue specifically for DiagnosticMessages completely changes the behavior of the `DoIPConnection`. Since I do not see a use case for this apart from the discoverer I thus decided to place it here instead.

If you think it makes more sense to incorporate it in the general `DoIPConnection`, I can easily do so with a flag to separate the queues in the `__init__` function?

Reply:
I think this would be a better solution. Then you have one entry point (= the DoipConnection class) that could be configured to behave in a certain way.

Reply:
I placed fixme commits on top. Please review whether this is to your liking!